src/mlagility/analysis/status.py

```diff
@@ -16,96 +16,147 @@ def update(models_found: Dict[str, ModelInfo]) -> None:
+                    script_names_visited=script_names_visited,
+                )
+
+
+def print_workload(
+    model_info: ModelInfo,
+    workload_hash: Union[str, None],
+    print_file_name: bool = False,
+    workload_idx: int = 0,
+    model_visited: bool = False,
 ) -> None:
     """
     Print information about a given model or submodel
     """
+    workload = model_info.workloads[workload_hash]
     ident = "\t" * (2 * model_info.depth + 1)
     if print_file_name:
         print(f"{model_info.script_name}.py:")
-    printing.log(f"{ident}{model_info.name} ")
 
-    # Show the number of times the model has been executed
-    # Only show the execution time if we are not running benchit() as this
-    # impacts time measurement.
-    if model_info.exec_time == 0 or model_info.build_model:
+    if workload.exec_time == 0 or model_info.build_model:
         exec_time = ""
     else:
-        exec_time = f" - {model_info.exec_time:.2f}s"
-    printing.logn(
-        f"(executed {model_info.executed}x{exec_time})",
-        c=printing.Colors.OKGREEN,
-    )
+        exec_time = f" - {workload.exec_time:.2f}s"
+
+    if model_info.depth == 0 and len(model_info.workloads) > 1:
+        if not model_visited:
+            printing.logn(f"{ident}{model_info.name}")
+    else:
+        printing.log(f"{ident}{model_info.name}")
+        printing.logn(
+            f" (executed {workload.executed}x{exec_time})",
+            c=printing.Colors.OKGREEN,
+        )
+
+    if (model_info.depth == 0 and not model_visited) or (model_info.depth != 0):
+        if model_info.depth == 0:
+            if model_info.model_type == build.ModelType.PYTORCH:
+                print(f"{ident}\tModel Type:\tPytorch (torch.nn.Module)")
+            elif model_info.model_type == build.ModelType.KERAS:
+                print(f"{ident}\tModel Type:\tKeras (tf.keras.Model)")
+
+        # Display class of found model and the file where it was found
+        model_class = type(model_info.model)
+        print(f"{ident}\tClass:\t\t{model_class.__name__} ({model_class})")
+        if model_info.depth == 0:
+            print(f"{ident}\tLocation:\t{model_info.file}, line {model_info.line}")
+
+        # Converting number of parameters to MB assuming 2 bytes per parameter
+        model_size = model_info.params * 2 / (1024 * 1024)
+        model_size = "{:.1f}".format(model_size) if model_size > 0.1 else "<0.1"
+        print(
+            f"{ident}\tParameters:\t{'{:,}'.format(model_info.params)} ({model_size} MB)"
+        )
 
-    if model_info.model_type == build.ModelType.PYTORCH:
-        print(f"{ident}\tModel Type:\tPytorch (torch.nn.Module)")
-    elif model_info.model_type == build.ModelType.KERAS:
-        print(f"{ident}\tModel Type:\tKeras (tf.keras.Model)")
+    if model_info.depth == 0 and len(model_info.workloads) > 1:
+        printing.logn(
+            f"\n{ident}\tWorkload {workload_idx+1} (executed {workload.executed}x{exec_time})",
```

Comment:
Since we are using the term "workload" to mean something very specific, could you add it to the tools user guide `Definitions` section? https://github.com/groq/mlagility/blob/main/docs/tools_user_guide.md#definitions

Reply:
Added and changed `workload` to `model` where appropriate in the documentation. This required changing many documents. Let me know if you find those changes appropriate or if you would rather revert them.